### PR TITLE
docs: Rewrite everything

### PR DESCRIPTION
**Pull request by jku**
I've done a major refactor of the documentation, would be great to get feedback from others: do you think this direction makes sense.

---

Link to new docs README: https://github.com/jku/tuf-on-ci/blob/docs-refactor/README.md. The structure is now roughly like this:

**level 0:**
* README: The elevator pitch only

**level 1 (manuals linked from README):**
* docs/SIGNER-MANUAL.md: how to sign and accept invites, nothing else
* docs/REPOSITORY-MAINTENANCE.md: How to setup a repo and modify delegations

**level 2 (linked from the manuals):**
* docs/SIGNER-SETUP.md: One-time setup for signers
* docs/ONLINE-SIGNING-SETUP.md: KMS specific details

**level 3:**
* docs/DEVELOPMENT.md: tips and tricks

### CODE REVIEW

**Comment by jku** [comment]
I seem to have dropped the note about `create-config-file.sh`, should add that back

**Comment by jku** [comment]
Things that I think are missing but I don't plan to add in this  PR:
* README should have a high level diagram and a very short description of how the system operates  
* More details about what "delegation" means somewhere
* more examples of specific actions ("removing a signer", "adding a new delegated role")

File: docs/SIGNER-MANUAL.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+_Role_: A role manages a set of artifacts and (optionally) a set of delegations to other
+roles. A role has a set of _signers_ (defined by the delegating role): their signatures
+are needed when the role is changed.
+The default delegation structure includes only a `root` role and a `targets`
+role (delegated by root). targets role can further delegate to other roles.  
+
+## Usage
+
+Metadata is signed in a _signing event_. Signing event process is:
```
Comment (lukpueh) [inline docs/SIGNER-MANUAL.md:29]:
```suggestion
Metadata is signed in a _signing event_. The signing event process is:
```

File: docs/SIGNER-MANUAL.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+* A signing event GitHub issue gets created by the repository. This happens as a
+  response to either a timed event (like an expiry date approaching) or as a response to
+  artifact changes: either way, the signing event contains new metadata versions that need
+  to be signed before they are considered valid.
+* Signing event directs _signers_ to sign the changes using `tuf-on-ci-sign`. By signing
+  they confirm that the proposed changes are correct. The signing tool updates the signing
+  event branch with the signatures
+  * If signers are not GitHub repository maintainer, they also create PRs from their fork
+    to the signing event
```
Comment (lukpueh) [inline docs/SIGNER-MANUAL.md:38]: Maybe to clarify:

```suggestion
  they confirm that the proposed changes are correct. The local signing tool updates the remote signing
  event branch with the signatures.
  * If a signer does not have push permissions for the GitHub repository, their signature is added to the signing event via PR from their fork.
```

File: docs/SIGNER-MANUAL.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+## Usage
+
+Metadata is signed in a _signing event_. Signing event process is:
+* A signing event GitHub issue gets created by the repository. This happens as a
+  response to either a timed event (like an expiry date approaching) or as a response to
+  artifact changes: either way, the signing event contains new metadata versions that need
+  to be signed before they are considered valid.
+* Signing event directs _signers_ to sign the changes using `tuf-on-ci-sign`. By signing
+  they confirm that the proposed changes are correct. The signing tool updates the signing
+  event branch with the signatures
```
Comment (lukpueh) [inline docs/SIGNER-MANUAL.md:36]: Consistency nit:
```suggestion
  event branch with the signatures.
```

File: docs/SIGNER-MANUAL.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+  * If signers are not GitHub repository maintainer, they also create PRs from their fork
+    to the signing event
+* Finally a Pull Request to merge the signing event into main is created
```
Comment (lukpueh) [inline docs/SIGNER-MANUAL.md:39]: Punctuation and consistency nit:
```suggestion
* Finally, a Pull Request to merge the signing event into main is created.
```

File: docs/SIGNER-MANUAL.md [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+but *signers should only ever fully trust their local signing tool*.
+
+The signing tool works in the repository (git clone) directory -- note that
+fetching, pushing or switching branches is not necessary: the tool will always use an
+up-to-date signing event branch and when the signer decides to sign, the signature is
+automatically pushed to the signing event branch.
+
+### Accepting an invitation 
+
+When a signing event instructs to accept an invitation to become a signer:
```
Comment (lukpueh) [inline docs/SIGNER-MANUAL.md:52]:
```suggestion
When a signing event invites to become a signer:
```
?

File: docs/SIGNER-MANUAL.md [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+
+The role where the artifact belongs to is chosen with pathname: 
+* files in the targets directory are artifacts managed by top level role "targets" 
+* files in a subdirectory are artifacts of the role with the same name (so
+  `targets/A/file.txt` is an artifact managed by role "A")
+
+<details>
+  <summary>Example</summary>
+
+  First, artifact changes are done with git to a signing event branch:
```
Comment (lukpueh) [inline docs/SIGNER-MANUAL.md:87]:
```suggestion
  Artifact changes are committed into a signing event branch using git:
```

**Review by lukpueh — CHANGES_REQUESTED** [review]
Awesome docs! This is a pure text review. I can do another pass, where I actually try it out.

**Comment by jku** [comment]
Thanks for review: I'll accept these in web ui and then likely do a rebase to have less commits

File: docs/SIGNER-MANUAL.md (hunk 5 above)
Comment (jku) [reply docs/SIGNER-MANUAL.md:51]: Tricky sentence. I'll make it _When a signing event GitHub issue invites to become a signer_ but even that is a bit ambiguous...

File: docs/REPOSITORY-MAINTENANCE.md [hunk 7]
```diff
@@ -0,0 +1,85 @@
+# TUF-on-CI Repository Maintenance Manual
+
+This page documents the initial setup of a TUF-on-CI repository as well as the
+ongoing maintenance.
+
+## New Repository Setup
+
+1. Fork the [template](https://github.com/theupdateframework/tuf-on-ci-template).
+1. To enable repository publishing, set _Settings->Pages->Source_ to `Github Actions`
+1. Clone the repository locally and [configure your signer tool](SIGNER-SETUP.md)
```
[the remark this answers is not in the document]
Comment (jku) [reply docs/REPOSITORY-MAINTENANCE.md:10]: this is tricky... I'm not 100% sure "offline signing" is something that people understand (especially considering tuf-on-ci-sign won't really work offline). Maybe well call it "local signing tool" more often to make the point?

File: docs/REPOSITORY-MAINTENANCE.md [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+1. Run `tuf-on-ci-delegate sign/init` to configure the repository and to start the
+   first signing event
+   * The tool prompts for various repository details and finally prompts to 
+     sign and push the initial metadata to a signing event branch
+1. when this initial signing event branch is merged, the repository generates the
+   first snapshot and timestamp, and publishes the first repository version
+
+## Modifying roles and creating new ones
+
+The same process handles:
```
[the remark this answers is not in the document]
Comment (jku) [reply docs/REPOSITORY-MAINTENANCE.md:23]: It was supposed to mean "all the cases below are handled in the same way". I will reword

**Review by kommendorkapten — APPROVED** [review]
Nice work! I like this layout.

**Pull request merged**